src/CommandIsolator.cpp

```diff
@@ -0,0 +1,64 @@
+#include "CommandIsolator.hpp"
+#include "CommandRunner.hpp"
+#include "Helpers.hpp"
+
+namespace criteo {
+namespace mesos {
+
+using std::string;
+
+using ::mesos::ContainerID;
+using ::mesos::slave::ContainerConfig;
+using ::mesos::slave::ContainerLaunchInfo;
+
+using process::Failure;
+
+CommandIsolator::CommandIsolator(
+  const string& prepareCommand,
+  const string& cleanupCommand)
+  : m_prepareCommand(prepareCommand),
+    m_cleanupCommand(cleanupCommand)
+{
+}
+
+process::Future<Option<ContainerLaunchInfo>> CommandIsolator::prepare(
+  const ContainerID& containerId,
+  const ContainerConfig& containerConfig) {
+  if(m_prepareCommand.empty()) return None();
+
+  LOG(INFO) << "prepare: calling command \"" << m_prepareCommand << "\"";
```

Comment:
Would be nice to have some context info as well such are the containerId

Reply:
I agree, I propose to add debug logs with the complete inputs in another review. It will be done for the isolator AND the hook.